multiaccounts/accounts/database.go

```diff
@@ -49,44 +91,196 @@ func (db Database) Close() error {
 	return db.db.Close()
 }
 
-func (db *Database) SaveConfig(typ string, value interface{}) error {
-	_, err := db.db.Exec("INSERT OR REPLACE INTO settings (type, value) VALUES (?, ?)", typ, &sqlite.JSONBlob{value})
-	return err
-}
+func (db *Database) CreateSettings(s Settings, nodecfg params.NodeConfig) error {
+	_, err := db.db.Exec(`
+INSERT INTO settings (
+  address,
+  currency,
+  current_network,
+  dapps_address,
+  eip1581_address,
+  installation_id,
+  key_uid,
+  keycard_instance_uid,
+  keycard_paired_on,
+  keycard_pairing,
+  latest_derived_path,
+  mnemonic,
+  name,
+  networks,
+  node_config,
+  photo_path,
+  preview_privacy,
+  public_key,
+  signing_phrase,
+  wallet_root_address,
+  synthetic_id
+) VALUES (
+?,?,?,?,?,?,?,?,?,?,
+?,?,?,?,?,?,?,?,?,?,
+'id')`,
+		s.Address,
+		s.Currency,
+		s.CurrentNetwork,
+		s.DappsAddress,
+		s.EIP1581Address,
+		s.InstallationID,
+		s.KeyUID,
+		s.KeycardInstanceUID,
+		s.KeycardPAiredOn,
+		s.KeycardPairing,
+		s.LatestDerivedPath,
+		s.Mnemonic,
+		s.Name,
+		s.Networks,
+		&sqlite.JSONBlob{nodecfg},
+		s.PhotoPath,
+		s.PreviewPrivacy,
+		s.PublicKey,
+		s.SigningPhrase,
+		s.WalletRootAddress)
 
-func (db *Database) GetConfig(typ string, value interface{}) error {
-	return db.db.QueryRow("SELECT value FROM settings WHERE type = ?", typ).Scan(&sqlite.JSONBlob{value})
+	return err
 }
 
-func (db *Database) GetConfigBlob(typ string) (rst json.RawMessage, err error) {
-	return rst, db.db.QueryRow("SELECT value FROM settings WHERE type = ?", typ).Scan(&rst)
-}
+// nolint: gocyclo
+func (db *Database) SaveSetting(setting string, value interface{}) error {
+	var (
+		update *sql.Stmt
+		err    error
+	)
 
-func (db *Database) GetConfigBlobs(types []string) (map[string]json.RawMessage, error) {
-	// it expands number of bind vars to the number of types. sql interface doesn't allow to bypass it without modifying driver.
-	// expansion can be hidden in a function that modifies string but better to make it explicitly.
-	query := fmt.Sprintf("SELECT type, value FROM settings WHERE type IN (?%s)", strings.Repeat(",?", len(types)-1)) // nolint: gosec
-	args := make([]interface{}, len(types))
-	for i := range types {
-		args[i] = types[i]
+	switch setting {
```

Comment:
You can replace all this with some rules I believe:
```go
func normalizeSettingName(name string) string {
    name = strings.ReplaceAll(name, "?", "")
    name = strings.ReplaceAll(name, "-", "_")
    // other corner cases
    return name
}

columnName := normalizeSettingName(setting)
update, err := db.db.Prepare(
    fmt.Sprintf("UPDATE settings SET %s = ? WHERE synthetic_id = 'id'", columnName)
)
if err != nil {
    return err
}
_, err = update.Exec(toSQLValue(name, value))
```

Reply:
I made this switch to avoid the SQL injection warning. It's fine like this

Reply:
If done with `fmt.Sprintf` before passing to `Prepare()`, there should be no SQL injection warnings. We can keep it this way, it's just a lot of manual work that computer could do.